### DISCUSSION TITLE
API routes developer and production environment differences

### DISCUSSION THREAD

**Discussion by thecoorum** [vercel/next.js#59652] [Help]
### Summary

Hey! I met an issue while configuring a Telegram bot using [Grammy](https://grammy.dev/) on API routes. While discussing the issue with the Grammy team we've came to the point that there may be differences between `Request`, `Response` and `Promise` implementation in local and production environments. While the whole implementation is working locally, the deployed one fails because of the function timeout (despite the timeout amount set in the `vercel.json`, I tried 15s and 300s).

Route implementation
```ts
import { NextRequest, NextResponse } from "next/server";
import { Bot, Composer, webhookCallback } from "grammy";

import { handlers } from "@/bot/handlers";

import { start } from "@/bot/commands/start";

import { supabase } from "@/utils/supabase";

import type { Context } from "@/bot/types";

export const POST = async (req: NextRequest, ...args: any[]) => {
  const { data, error } = await supabase
    .from("companies")
    .select("bot_token")
    .eq("slug", req.headers.get("host")!.split(".")[0])
    .single();

  if (error) {
    return NextResponse.json(
      {
        message: "Company not found",
      },
      { status: 404 }
    );
  }

  const token = data.bot_token;

  const middleware = new Composer<Context>();

  middleware.command("start", start);

  const bot = new Bot<Context>(token);

  bot.on("message", async (ctx) => {
    await ctx.reply("...");
  });
  bot.use(handlers);
  bot.use(middleware);

  const handleUpdate = webhookCallback(bot, "std/http");

  return handleUpdate(req, ...args);
};
```

Original issue on Grammy: https://github.com/grammyjs/grammY/issues/506#issuecomment-1857624689

Any suggestions will be appreciated! Thanks in advance!

**Comment by di-sukharev** [comment]
i think i've solved it https://github.com/grammyjs/grammY/issues/506#issuecomment-1891294941

**Comment by PonomareVlad** [comment]
- [x] https://github.com/grammyjs/examples/pull/47
